force-app/main/default/classes/plugin/EmailLoggerPlugin.cls

```diff
@@ -37,47 +40,45 @@ public without sharing class EmailLoggerPlugin implements LoggerPlugin.Triggerab
         //to avoid trigger recursion
         //update in line 62 causes trigger recursion
         if (isFirstExecution) {
-            List<Log__c> filtered = filterNotSentLogs(logs);
-            Set<Id> logIds = (new Map<Id, Log__c>(filtered)).keySet();
+            List<Log__c> filteredLogs = filterNotSentLogs(logs);
+            Set<Id> logIds = (new Map<Id, Log__c>(filteredLogs)).keySet();
             // requery logs with their log entries
-            List<Log__c> fullLogs = requeryLogs(logIds);
-            List<Messaging.SingleEmailMessage> emails = new List<Messaging.SingleEmailMessage>();
-            if (!EMAIL_RECIPIENTS.isEmpty()) {
-                String orgWideEmailName = LoggerParameter.getString('EmailLoggingOrgWideAddress', '');
-                Id orgWideEmailId = EmailLoggerUtils.getOrgWideEmailAddress(ORG_WIDE_EMAIL_NAME)?.Id;
+            List<Log__c> fullLogs = requeryFullLogs(logIds);
+            List<Messaging.SingleEmailMessage> emailLogs = new List<Messaging.SingleEmailMessage>();
+            if (!LOG_RECIPIENTS.isEmpty()) {
+                String defaultSenderName = LoggerParameter.getString('EmailLoggingOrgWideEmailAddressName', null);
+                Id defaultSenderId = EmailLoggerUtils.getOrgWideEmailAddress(defaultSenderName)?.Id;
```

Comment:
if defaultSenderId == null. Will it work or fail?

Reply:
It will work.